pytradfri/gateway.py

```diff
@@ -287,7 +288,10 @@ def update(self):
         def process_result(result):
             self.raw = result
 
-        return Command("get", self.path, process_result=process_result)
+        cmd = Command("get", self.path, process_result=process_result)
+        if not self.raw:
```

Comment:
I still don't think this is the right fix. Why are we checking for `self.raw` here? We don't use `self.raw` in this method. We need to find the source of the problem and address it there.

Reply:
There are 2 places in the entire lib where self.raw is assigned, those are now protected. The assignment self.raw = result is the source of the problem. Sometimes a Command fails (that seems to be very common with IKEA Gateway), and that leads to result being None. Now we can either check for self.raw None (like it is done in device) or raise an exception. The library is not prepared for automatic reissue of command, nor do I think it is a good idea, so the best solution is to report the problem to the calling app, and let it decide.

self.raw == None is the problem in a number of reported issues in HA, and I can now produce the problem here relatively easy.

Reply:
But we don't assign `self.raw` here. It's done in the nested function that we define above, which we pass as a callback to be called when processing this command. So checking here doesn't guard anything in this method before we return.

Reply:
I guess you're talking about the next commit.